tests/e2e/common/kubernetes.go

```diff
@@ -164,27 +165,38 @@ func CleanupNode(ctx context.Context, cs clientset.Interface) {
 	}
 }
 
-func CleanupNodeFeatureRules(ctx context.Context, cli *nfdclient.Clientset, namespace string) {
-	// Drop NodeFeatureRule objects
+func CleanupNodeFeatures(ctx context.Context, cli *nfdclient.Clientset, namespace string) {
+	// Clean NodeFeatureRule objects
 	nfrs, err := cli.NfdV1alpha1().NodeFeatureRules().List(ctx, metav1.ListOptions{})
 	Expect(err).NotTo(HaveOccurred())
 
 	if len(nfrs.Items) != 0 {
 		By("Deleting NodeFeatureRule objects from the cluster")
 		for _, nfr := range nfrs.Items {
```

Comment:
Question: Could we be deleting node feature rules that are NOT controlled by our components?